src/FirmwarePlugin/APM/APMFirmwarePlugin.cc

```diff
@@ -825,3 +827,84 @@ QString APMFirmwarePlugin::internalParameterMetaDataFile(Vehicle* vehicle)
         return QString();
     }
 }
+
+QString APMFirmwarePlugin::_getGitVersionUrl(Vehicle* vehicle)
+{
+    QString  baseUrl("http://firmware.ardupilot.org/%1/stable/PX4/git-version.txt");
+    if (vehicle->fixedWing()) {
+        return baseUrl.arg("Plane");
+    } else if (vehicle->multiRotor()) {
+        return baseUrl.arg("Copter");
+    } else if (vehicle->vtol()) {
+        return baseUrl.arg("Plane");
+    } else if (vehicle->rover()) {
+        return baseUrl.arg("Rover");
+    } else if (vehicle->sub()) {
+        return baseUrl.arg("Sub");
+    }
+    return QString("http://firmware.ardupilot.org/Copter/stable/PX4/git-version.txt");
+}
+
+void APMFirmwarePlugin::checkIfIsLatestStable(Vehicle* vehicle)
+{
+    // This is required as mocklink uses a hardcoded firmware version
+    if (qgcApp()->runningUnitTests()) {
+        qCDebug(VehicleLog) << "Skipping version check";
+        return;
+    }
+    QString versionFile = _getGitVersionUrl(vehicle);
+    qCDebug(VehicleLog) << "Downloading" << versionFile;
+    QGCFileDownload* downloader = new QGCFileDownload(this);
+    connect(
+        downloader,
+        &QGCFileDownload::downloadFinished,
+        [vehicle, this]( QString remoteFile, QString localFile) {
```

Comment:
The recommendation is to use the 4-arg overload, which takes a context object so that the lambda isn't executed when the context object is deleted.
The correct way is :
```
connect(
        downloader,
        &QGCFileDownload::downloadFinished,
        **this**,
        [vehicle, this]( QString remoteFile, QString localFile) 
```